benchmark/utils/formats.hpp

```diff
@@ -156,6 +159,29 @@ namespace formats {
 // some shortcuts
 using hybrid = gko::matrix::Hybrid<etype>;
 using csr = gko::matrix::Csr<etype>;
+using batch_csr = gko::matrix::BatchCsr<etype>;
+
+/**
+ * Creates a Ginkgo matrix from the intermediate data representation format
+ * gko::matrix_data.
+ *
+ * @param exec  the executor where the matrix will be put
+ * @param data  the data represented in the intermediate representation format
+ *
+ * @tparam MatrixType  the Ginkgo matrix type (such as `gko::matrix::Csr<>`)
+ *
+ * @return a `unique_pointer` to the created matrix
+ */
+template <typename MatrixType>
+std::unique_ptr<MatrixType> read_batch_matrix_from_data(
+    std::shared_ptr<const gko::Executor> exec, const int num_duplications,
+    const gko::matrix_data<etype> &data)
```

Comment:
Wouldn't it be more useful to take a vector of matrix_data (representing an entire batch) as input, and duplicate the entire batch?